pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -660,12 +660,14 @@ func getNodeIDAsHexFromXfrmMark(mark *netlink.XfrmMark) string {
 }
 
 func getDirFromXfrmMark(mark *netlink.XfrmMark) dir {
-	switch {
-	case mark == nil:
+	if mark == nil {
 		return dirUnspec
-	case mark.Value&linux_defaults.RouteMarkDecrypt != 0:
+	}
+	bitwiseResult := mark.Value & linux_defaults.EncryptDecryptMask
```

Comment:
This should be the existing `linux_defaults.RouteMarkMask` (which we should probably rename as it's now specific to IPsec AFAIK).
```suggestion
	bitwiseResult := mark.Value & linux_defaults.RouteMarkMask
```

Reply:
I can take care of the renaming in another PR (to not make backports more complicated here).